cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -545,50 +593,51 @@ exports.certificateRequestHandler = async function (event, context) {
     OtherDomainZone: {},
   };
 
+  let certDomain = `${props.EnvName}.${props.AppName}.${props.DomainName}`;
+  let aliases = await getAllAliases(props.Aliases);
+  const uniqueSansToUse = new Set([certDomain, `*.${certDomain}`, ]);
+  for (const alias of aliases) {
+    uniqueSansToUse.add(alias);
+  }
+  const sansToUse = [...uniqueSansToUse, ];
+  const [acm, envRoute53, appRoute53] = clients(props.Region, props.RootDNSRole);
+
   try {
-    var certDomain = `${props.EnvName}.${props.AppName}.${props.DomainName}`;
-    var aliases = await getAllAliases(props.Aliases);
+    let response = {};
+    let options = {};
     switch (event.RequestType) {
       case "Create":
-        certificateArn = await requestCertificate(
+        response = await requestCertificate(
           event.RequestId,
           props.AppName,
           props.EnvName,
           certDomain,
           aliases,
-          props.EnvHostedZoneId,
-          props.RootDNSRole,
-          props.Region
+          sansToUse,
+          acm
         );
-        responseData.Arn = physicalResourceId = certificateArn;
+        responseData.Arn = physicalResourceId = response.CertificateArn;  // Set physicalResourceId as soon as we can.
+        options = await waitForValidationOptionsToBeReady(response.CertificateArn, sansToUse, acm);
+        await validateCertificate(options, envRoute53, appRoute53, props.EnvHostedZoneId, response.CertificateArn, acm);
```

Comment:
oh i see, so now we separated the certificate request logic compared to before where we would merge all three operations into one place